jscomp/core/js_fold_basic.ml

```diff
@@ -49,7 +49,7 @@ let needBlockRuntimeInReleaseMode (tag : J.expression) (tag_info : J.tag_info) =
   | Blk_record _  
   | Blk_tuple 
   | Blk_array -> false   
-  | Blk_constructor (_, 1)      
+  | Blk_constructor (_, 1, _)      
```

Comment:
Blk_some seems  need runtime support for most cases (also need check debug mode)